guides/plugins/plugins/content/cms/real-world-cms/add-real-world-cms-block.md

```diff
@@ -0,0 +1,268 @@
+# Add a real world CMS block
+
+## Overview
+
+This guide will teach you how to create a CMS block with your plugin. This guide will cover a real world example to keep everything easy we will just create a  button. The CMS block for the button we will cover right here.
+
+## Prerequisites
+
+This plugin is built based on the [Add CMS block](../add-cms-block.md) guide. We assume that you know the basics for a CMS block from this guide.
+
+## Custom block in the administration
+
+Let's get started with adding the custom block for a button in the administration. Since Shopware 6 already comes with different categories we just have to choose in which category we want to show our block. If you break down what a button is, you should come to the conclusion that a button is just text. So we create the block in the `text` category.
+
+## Administration code
+
+Since you already know the basics from [Add CMS block](../add-cms-block.md), we will just look a quick look at the code.
+First we create our `main.js` in the  `<plugin root>/src/Resources/app/administration/src` directory.
+
+{% code title="<plugin root>/src/Resources/app/administration/src/main.js" %}
+```javascript
+import './module/sw-cms/blocks/text/cms-button';
+```
+{% endcode %}
+
+### Block category
+We decided to create our block in the text category, so we place our first `index.js` in `<plugin root>/src/Resources/app/administration/src/module/sw-cms/blocks/text`
+
+{% code title="<plugin root>/src/Resources/app/administration/src/module/sw-cms/blocks/text/cms-button/index.js" %}
+```javascript
+import './component';
+import './preview';
+
+Shopware.Service('cmsService').registerCmsBlock({
+    name: 'cms-button',
+    label: 'sw-cms.blocks.text.ninja-cms-button.label',
+    category: 'text',
+    component: 'sw-cms-block-button',
+    previewComponent: 'sw-cms-preview-button',
+    defaultConfig: {
+        marginBottom: '20px',
+        marginTop: '20px',
+        marginLeft: '20px',
+        marginRight: '20px',
+        sizingMode: 'boxed'
+    },
+    /* The slot config is important!
+       We give the slot button the element text in this tutorial because we haven't defined a button element yet.
+       You can use the existing blocks and elements out of the box but it gets interesting if you want to create your very own element with an own config.
+       We'll take a look at this later in this guide.
```

Comment:
```suggestion
       We give the slot button the element text in this tutorial because we haven't defined a button element yet.
       You can use the existing blocks and elements out of the box, but it gets interesting if you want to create your own element with a custom config.
       We'll take a look at this later in this guide.
```